test/integration/ProjectBrowserTest.kt

```diff
@@ -36,103 +39,115 @@ class ProjectBrowserTest : IntegrationTest() {
     fun createADirectory() {
         runBlocking {
             val createdDirectoryName = "created"
-
-            copyFolder(source = SampleFileStructure.path, destination = testID)
+            val path = copyFolder(source = SampleFileStructure.path, destination = testID)
             openProject(composeRule, projectDirectory = testID)
 
-            Service.project.current!!.directory.asDirectory()
-                .tryCreateDirectory(createdDirectoryName)
-            delayAndRecompose(composeRule)
+            Service.project.current!!.directory.asDirectory().tryCreateDirectory(createdDirectoryName)
+
+            val file = File("$path/$createdDirectoryName")
+            StudioActions.waitUntilAssertionIsTrue(composeRule) {
+                file.exists()
+            }
 
             Service.project.current!!.reloadEntries()
-            delayAndRecompose(composeRule)
 
-            assertNodeExistsWithText(composeRule, text = createdDirectoryName)
+            waitUntilNodeWithTextExists(composeRule, text = createdDirectoryName)
```

Comment:
This definitely seems like a call that should have the word "assert" in it somewhere.

Reply:
We're just asserting after. Although it's a little bit of duplication, it reads better in my opinion.